ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/DB/AcidState.hs

```diff
@@ -426,23 +446,18 @@ createAccPrefiltered mkPrefilteredUtxo accApplyP narrowP addrApplyP accId p mbAc
             , _accCheckpointBlockMeta   = mempty
             }
 
-        newAddress :: HdAddressId -> Core.Address -> Utxo -> HdAddress
-        newAddress addressId address addrUtxo =
-            HD.initHdAddress addressId (InDb address) (firstAddrCheckpoint addrUtxo)
-
-        firstAddrCheckpoint :: Utxo -> AddrCheckpoint
-        firstAddrCheckpoint addrUtxo = AddrCheckpoint {
-              _addrCheckpointUtxo        = InDb addrUtxo
-            , _addrCheckpointUtxoBalance = InDb $ Spec.balance addrUtxo
-            }
-
 {-------------------------------------------------------------------------------
   Wrap HD C(R)UD operations
 -------------------------------------------------------------------------------}
-
-createHdAddress :: HdAddress -> Update DB (Either HD.CreateHdAddressError ())
-createHdAddress hdAddress = runUpdate' . zoom dbHdWallets $
+createHdAddress :: HdAddressId
+                -> Core.Address
+                -> AddrCheckpoint
+                -> Update DB (Either HD.CreateHdAddressError HdAddress)
+createHdAddress addressId newAddress firstCheckpoint = runUpdate' . zoom dbHdWallets $ do
+    let hdAddress :: HdAddress
+        hdAddress = HD.initHdAddress addressId (InDb newAddress) firstCheckpoint
     HD.createHdAddress hdAddress
+    pure hdAddress
```

Comment:
The idea was to use `createAddPrefiltered` in this function. Currently this refactoring doesn't make much sense, you added a subfunction of `createAccPrefiltered` and use it only in `createAccPrefiltered`.